src/Http/Http.Results/src/Results.cs

```diff
@@ -595,18 +595,24 @@ public static IResult StatusCode(int statusCode)
     /// <summary>
     /// Produces a <see cref="StatusCodes.Status404NotFound"/> response.
     /// </summary>
-    /// <param name="value">The value to be included in the HTTP response body.</param>
     /// <returns>The created <see cref="IResult"/> for the response.</returns>
-    public static IResult NotFound(object? value = null)
-        => NotFound<object>(value);
+    public static IResult NotFound()
```

Comment:
These are breaking changes to the public API and are unlikely to be accepted.

Reply:
The linter obligated me to remove this implementation, I will undo this specific change and look for a different solution. I see this breaks non inferred type calls.

Reply:
@martincostello, the only case where this would be a breaking change is if someone try to use it like this:

> Results.NotFound(null)

I do not see this being a use case, should this actually be undone? If this is to be undone, there will be multiple overloads with optional parameters. [[RS0026]](https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md)

Reply:
Anything that requires you to change any of the PublicAPI.**shipped**.txt files is a breaking change. Those files are effectively a snapshot and considered a baseline that shouldn't be changed. The linter was telling you the changes were breaking as it could no longer find the members in the file (because you changed the method signature, which is binary-breaking).

Changes that add new public API need to go through the API review process after an API proposal, which there's an issue template for here: https://github.com/dotnet/aspnetcore/issues/new/choose

Reply:
> Anything that requires you to change any of the PublicAPI.**shipped**.txt files is a breaking change. Those files are effectively a snapshot and considered a baseline that shouldn't be changed. The linter was telling you the changes were breaking as it could no longer find the members in the file (because you changed the method signature, which is binary-breaking).
> 
> Changes that add new public API need to go through the API review process after an API proposal, which there's an issue template for here: https://github.com/dotnet/aspnetcore/issues/new/choose

I understand that this is a breaking change because of the changes on the PublicAPI.shipped.txt file, but the linter was not warning me about the breaking changes, the linter was asking me to not make two method overloads with optional parameters, its the [RS0026](https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md) lint rule.

![image](https://github.com/dotnet/aspnetcore/assets/68293882/9fa0ed4e-9ad4-4ee9-9828-32db67d3fc79)

Please ignore the [RS0016] warn on the image, it is there because I just remade the method as it was to demonstrate the problem with the image.

Reply:
Sure, but by fixing RS0026 rule you've introduced a breaking change. The linter is no longer complaining because you've deleted the original method signature and any record of it from the Shipped file, and then added a new method signature to the Unshipped file - at this point the linter can no longer warn about the breaking change because it's no longer visible to it.

You should raise an API proposal, which if approved, will determine how to proceed, whether that is suppressing the RS0026 warning, using a different method name, or something else.

At that point you can update this PR to adopt whatever the shape of any approved APIs are.

Reply:
@martincostello Right, I will revert the changes to the shipped public api, if I find of a way of implementing this without affecting the shipped public method signatures I will notify you